### PR TITLE
feat: add get flights interface

### PR DESCRIPTION
**Pull request by ServiceDog**

### CODE REVIEW

**Comment by ServiceDog** [comment]
@owlot I can put in a svc-discovery and svc-telemetry PR when this goes in

File: server/src/postgis/flight.rs [hunk 1]
```diff
@@ -346,11 +350,207 @@ pub async fn get_flight_intersection_stmt(
...
+    let stmt = client
+        .prepare_cached(&format!(
+            r#"SELECT
+                    "geom",
+                    "velocity_horizontal_ground_mps",
+                    "velocity_vertical_mps",
+                    "track_angle_degrees",
+                    "last_position_update",
+                    "op_status"
+                FROM {table_name}
```
Comment (owlot) [inline server/src/postgis/flight.rs:483]: table name quotes

File: server/src/postgis/flight.rs [hunk 2]
```diff
@@ -346,11 +350,207 @@ pub async fn get_flight_intersection_stmt(
...
+            "(get_flights) could not get client from psql connection pool: {}",
+            e
+        );
+        FlightError::Client
+    })?;
+
+    let stmt = client
+        .prepare_cached(&format!(
+            r#"
+            SELECT
```
Comment (owlot) [inline server/src/postgis/flight.rs:408]: Query is missing table name quotes. Especially the ones on the variable names

File: server/src/postgis/flight.rs [hunk 3]
```diff
@@ -346,11 +350,207 @@ pub async fn get_flight_intersection_stmt(
...
+        "#,
+            table_name = super::aircraft::get_table_name(),
+        ))
+        .await
+        .map_err(|e| {
+            postgis_error!("(get_flights) could not prepare cached statement: {}", e);
+            FlightError::DBError
+        })?;
+
+    fn process_row(
```
Comment (owlot) [inline server/src/postgis/flight.rs:495]: I don't think it's a good idea to rely on column order to get the row data.
You can get the data using the column name instead (see example in svc_storage: https://github.com/Arrow-air/svc-storage/blob/8baec86c3ae5233a36210810ccc5226cfa323a64/server/src/resources/flight_plan/mod.rs#L217)

File: server/src/postgis/flight.rs (hunk 1 above)
Comment (ServiceDog) [reply server/src/postgis/flight.rs:483]: I use the get_table_name() getter which returns the quoted table in format "schema"."table" (with quotes), is this sufficient?

or should I instead separately insert schema and table into this format string with the quotes like "{schema}"."{table}"?

File: server/src/postgis/flight.rs (hunk 3 above)
Comment (ServiceDog) [reply server/src/postgis/flight.rs:495]: Great call, forgot this was possible

File: server/src/postgis/flight.rs (hunk 1 above)
Comment (owlot) [reply server/src/postgis/flight.rs:483]: Ah, yeah, it makes it a bit hard to determine it's being quoted, but it's all good then

File: server/src/postgis/flight.rs (hunk 3 above)
Comment (ServiceDog) [reply server/src/postgis/flight.rs:499]: https://github.com/Arrow-air/svc-gis/pull/19/commits/208ea9f19d9c58fd9d971792242d17f2a1b0b87e

File: server/src/postgis/flight.rs (hunk 2 above)
Comment (ServiceDog) [reply server/src/postgis/flight.rs:412]: https://github.com/Arrow-air/svc-gis/pull/19/commits/208ea9f19d9c58fd9d971792242d17f2a1b0b87e

**Review by owlot — APPROVED** [review]

**Pull request merged**